### ISSUE TITLE
Crash with golang 1.23.2

### ISSUE THREAD

**Issue by halsten-dev** [harryjph/go-common-file-dialog#11]
Hi,

Your lib is amazing and work great with golang 1.22.8.

Sadly it crashes completely with 1.23.3. The sysCall fails. I have not the knowledge to find a solution but the error looks exactly like the other issues declared in the past.

Could you take a look ? Or maybe give an hint on the reason why this happens ?

Thanks a lot in advance.

Have a wonderful day / evening / night

**Comment by karelbilek** [comment]
I have the same problem with go 1.24

**Comment by karelbilek** [comment]
It's this issue
https://github.com/golang/go/issues/69029

> This regression you are seeing when using go1.23 is probably related to [CL 563315](https://go-review.googlesource.com/c/go/+/563315), which optimized the syscall.Syscall performance and reduced the stack usage by using the nargs parameter, which was previously ignored.

**Comment by karelbilek** [comment]
Yep, either fixing the nargs or (better) switching to SysCallN fixes the issue

**Comment by harryjph** [comment]
I've merged this change, so this should hopefully be fixed now 😄

**Comment by halsten-dev** [comment]
Amazing ! Thanks for your analyse and work @karelbilek ! 👍

**Comment by karelbilek** [comment]
@halsten-dev the actual hard work was done by @leaanthony who uses this library (just "vendored" inside) in Wails

**Comment by williambrode** [comment]
Thanks for handling this!  I just hit the crash and need the fix too.

**Comment by leaanthony** [comment]
> [@halsten-dev](https://github.com/halsten-dev) the actual hard work was done by [@leaanthony](https://github.com/leaanthony) who uses this library (just "vendored" inside) in [Wails](https://github.com/wailsapp/wails)

Thanks for taking the time to push this upstream. I'm not sure why we vendored it in the first place. If it's being actively supported I'm happy to switch back to using this version.

**Comment by williambrode** [comment]
Should a new release be created so users can easily get this fix when updating go modules?
